core-blocks/gallery/index.js

```diff
@@ -125,12 +125,15 @@ export const settings = {
 				},
 			},
 			{
+				// When created by drag and dropping multiple files on an insertion point
 				type: 'files',
 				isMatch( files ) {
 					return files.length !== 1 && every( files, ( file ) => file.type.indexOf( 'image/' ) === 0 );
 				},
 				transform( files, onChange ) {
-					const block = createBlock( 'core/gallery' );
+					const block = createBlock( 'core/gallery', {
+						images: files.map( ( file ) => ( { url: window.URL.createObjectURL( file ) } ) ),
```

Comment:
It won't work out of the box, since you reference `window` object which isn't present in React Native. I wouldn't worry about it at the moment because drag and drop isn't supported there.